fineract-provider/src/main/java/org/apache/fineract/useradministration/starter/UserAdministrationAutoConfiguration.java

```diff
@@ -0,0 +1,101 @@
+                appUserRepository, staffReadPlatformService);
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    public AppUserWritePlatformServiceJpaRepositoryImpl appUserWritePlatformServiceJpaRepository(PlatformSecurityContext context,
+            UserDomainService userDomainService, PlatformPasswordEncoder platformPasswordEncoder, AppUserRepository appUserRepository,
+            OfficeRepositoryWrapper officeRepositoryWrapper, RoleRepository roleRepository, UserDataValidator fromApiJsonDeserializer,
+            AppUserPreviousPasswordRepository appUserPreviewPasswordRepository, StaffRepositoryWrapper staffRepositoryWrapper,
+            ClientRepositoryWrapper clientRepositoryWrapper) {
+        return new AppUserWritePlatformServiceJpaRepositoryImpl(context, userDomainService, platformPasswordEncoder, appUserRepository,
+                officeRepositoryWrapper, roleRepository, fromApiJsonDeserializer, appUserPreviewPasswordRepository, staffRepositoryWrapper,
+                clientRepositoryWrapper);
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    public PasswordPreferencesWritePlatformServiceJpaRepositoryImpl passwordPreferencesWritePlatformServiceJpaRepository(
+            PasswordValidationPolicyRepository validationPolicyRepository, PasswordPreferencesDataValidator dataValidator) {
+        return new PasswordPreferencesWritePlatformServiceJpaRepositoryImpl(validationPolicyRepository, dataValidator);
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    public PasswordValidationPolicyReadPlatformServiceImpl passwordValidationPolicyReadPlatformService(JdbcTemplate jdbcTemplate,
+            DatabaseSpecificSQLGenerator sqlGenerator) {
+        return new PasswordValidationPolicyReadPlatformServiceImpl(jdbcTemplate, sqlGenerator);
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    public PermissionReadPlatformServiceImpl permissionReadPlatformService(PlatformSecurityContext context, JdbcTemplate jdbcTemplate,
+            DatabaseSpecificSQLGenerator sqlGenerator) {
+        return new PermissionReadPlatformServiceImpl(context, jdbcTemplate, sqlGenerator);
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    public PermissionWritePlatformServiceJpaRepositoryImpl permissionWritePlatformServiceJpaRepository(PlatformSecurityContext context,
+            PermissionRepository permissionRepository, PermissionsCommandFromApiJsonDeserializer fromApiJsonDeserializer) {
+        return new PermissionWritePlatformServiceJpaRepositoryImpl(context, permissionRepository, fromApiJsonDeserializer);
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    public RoleReadPlatformServiceImpl roleReadPlatformService(JdbcTemplate jdbcTemplate) {
+        return new RoleReadPlatformServiceImpl(jdbcTemplate);
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    public RoleWritePlatformServiceJpaRepositoryImpl roleWritePlatformServiceJpaRepository(PlatformSecurityContext context,
```

Comment:
Don't use implementation classes here, always use Java interfaces... they are not excheangable! Please provide the Java interface class reference in parenthesis with the `@ConditionalOnMissingBean` annotation.

Reply:
Done.